File: subsystem_cpp/subsystem/sdk/cviewcontroller.cpp
#include "cviewcontroller.h"

struct _self_CViewController {
    CView::ref view;
};

CViewController::CViewController() {
    C_INIT(self);
}

void CViewController::viewDidLoad() {
}

void CViewController::viewDidAppear() {
}

void CViewController::viewDidDisappear() {
}

CView::ref CViewController::view() {
    if (self->view == nullptr) {
        loadView();
    }
    return self->view;
}

void CViewController::loadView() {
    self->view = CView::create();
}

File: subsystem_lua/subsystem/sdk/cviewcontroller.lua
CViewController = extends(CObject, {
    view = nil,
})

function CViewController:init()
    CViewController.super.init(self)
end

function CViewController:viewDidLoad()
end

function CViewController:viewDidAppear()
end

function CViewController:viewDidDisappear()
end

function CViewController:view()
    if self.view == nil then
        self:loadView()
    end
    return self.view
end

function CViewController:loadView()
    self.view = CView:new()
end
